### ISSUE TITLE
False Negative: UrF missing for unread fields in Enum

### ISSUE THREAD

**Issue by leemeii** [spotbugs/spotbugs#3955]
**Describe the bug**
Tool fails to report UrF: Unread field for a package-private field defined directly within an enum, even though the field is never read.

**To Reproduce**

```java
enum A {
  ENUM_VALUE_3;
  class B {
    int s = 0;       // <- reported UrF (TP)
  }
  boolean i = false; // <- should report UrF (FN)
}
```

**Expected behavior**
Both A$B.s and A.i should be reported with UrF: Unread field. Since both are package-private and neither is ever accessed or read by any other part of the code, they both represent dead code/redundant fields.

**Actual behavior**
Only the field s inside the inner class B is reported. The field i in the enum A is silently ignored.

**Output:**

Plaintext
M P UrF: Unread field: A$B.s  At InputInnerTypeLastClassRootClass2.java:[line 4]

**Environment**
- Java version: JDK 21
- SpotBugs Version: 4.9.8
- How do you use SpotBugs:  CLI

**Comment by StardustEnigma** [comment]
Hi, I’d like to work on this issue if no one else is currently working on it. I was able to reproduce the problem locally and will investigate a fix.

**Comment by ThrawnCA** [comment]
> Hi, I’d like to work on this issue if no one else is currently working on it.

Feel free.
